### ISSUE TITLE
Error: could not retrieve proofs mode: 'getProofsMode'

### ISSUE THREAD

**Issue by carsonfly** [filecoin-project/go-filecoin#3021]
### Description
When I trying to mine,I meat a problem.
When I type:
 go-filecoin miner create 100 --gas-price=0.001 --gas-limit=300 --peerid `go-filecoin id | jq -r '.ID'`   # this may take a minute
Throw Exception like :
Error: could not retrieve proofs mode: 'getProofsMode' query message failed: querymethod returned an error: failed to get To actor: actor not found

How can I solve it?

**Comment by travisperson** [comment]
Can you provide the output of `go-filecoin inspect all`?

**Comment by carsonfly** [comment]
Version:        81e75fd7f9722f9a100b8d03852e4e823a4b209d

Runtime
OS:             linux
Arch:           amd64
Version:        go1.12.1
Compiler:       gc
NumProc:        4
GoMaxProcs:     4
NumGoRoutines:  98
NumCGoCalls:    4

Disk
Free:   8706824503296
Total:  8267494649856
FSType: 61267

Memory
Swap:           0
Virtual:        9520944000

Environment
FilAPI:
FilPath:
GoPath:

Config
{
        "api": {
                "address": "/ip4/127.0.0.1/tcp/3453",
                "accessControlAllowOrigin": [
                        "http://localhost:8080",
                        "https://localhost:8080",
                        "http://127.0.0.1:8080",
                        "https://127.0.0.1:8080"
                ],
                "accessControlAllowCredentials": false,
                "accessControlAllowMethods": [
                        "GET",
                        "POST",
                        "PUT"
                ]
        },
        "bootstrap": {
                "addresses": [
                        "/dns4/user.kittyhawk.wtf/tcp/9000/ipfs/Qmd6xrWYHsxivfakYRy6MszTpuAiEoFbgE1LWw4EvwBpp4",
                        "/dns4/user.kittyhawk.wtf/tcp/9001/ipfs/QmXq6XEYeEmUzBFuuKbVEGgxEpVD4xbSkG2Rhek6zkFMp4",
                        "/dns4/user.kittyhawk.wtf/tcp/9002/ipfs/QmXhxqTKzBKHA5FcMuiKZv8YaMPwpbKGXHRVZcFB2DX9XY",
                        "/dns4/user.kittyhawk.wtf/tcp/9003/ipfs/QmZGDLdQLUTi7uYTNavKwCd7SBc5KMfxzWxAyvqRQvwuiV",
                        "/dns4/user.kittyhawk.wtf/tcp/9004/ipfs/QmZRnwmCjyNHgeNDiyT8mXRtGhP6uSzgHtrozc42crmVbg"
                ],
                "minPeerThreshold": 1,
                "period": "10s"
        },
        "datastore": {
                "type": "badgerds",
                "path": "badger"
        },
        "heartbeat": {
                "beatTarget": "/dns4/stats-infra.kittyhawk.wtf/tcp/8080/ipfs/QmUWmZnpZb6xFryNDeNU7KcJ1Af5oHy7fB9npU67sseEjR",
                "beatPeriod": "3s",
                "reconnectPeriod": "10s",
                "nickname": "Carson"
        },
        "mining": {
                "minerAddress": "empty",
                "autoSealIntervalSeconds": 120,
                "storagePrice": "0"
        },
        "mpool": {
                "maxPoolSize": 10000,
                "maxNonceGap": "100"
        },
        "net": "devnet-user",
        "observability": {
                "metrics": {
                        "prometheusEnabled": false,
                        "reportInterval": "5s",
                        "prometheusEndpoint": "/ip4/0.0.0.0/tcp/9400"
                },
                "tracing": {
                        "jaegerTracingEnabled": false,
                        "probabilitySampler": 1,
                        "jaegerEndpoint": "http://localhost:14268/api/traces"
                }
        },
        "sectorbase": {
                "rootdir": ""
        },
        "swarm": {
                "address": "/ip4/0.0.0.0/tcp/6000"
        },
        "wallet": {
                "defaultAddress": "t1zhfflggti7xaavdunzqe73dijhpukzd256tobvi"
        }
}

**Comment by carsonfly** [comment]
My peer ID is QmbuSbsJpcy2KoVHLEgiDDuFcAY77NqPEMNbSV7JvKzP67
Swarm listening on: /ip4/127.0.0.1/tcp/6000
Swarm listening on: /ip4/23.92.78.109/tcp/6000
Swarm listening on: /ip4/10.255.67.20/tcp/6000
22:37:35.411  INFO chain.stor: start loading chain at tipset: { zDPWYqFD3SJXwpUGq5HUVUiKaSAtmdBBUmD9QnxchzGkyiBHYsUT }, height: 0 store.go:111
22:37:35.412  INFO chain.stor: finished loading 0 tipsets from { zDPWYqFD3SJXwpUGq5HUVUiKaSAtmdBBUmD9QnxchzGkyiBHYsUT } store.go:155
**22:37:46.042 WARNI /fil/hello: failed to send hello handshake to peer QmUWmZnpZb6xFryNDeNU7KcJ1Af5oHy7fB9npU67sseEjR: protocol not supported hello.go:179**
22:37:46.187  INFO    metrics: successfully to open stream, peerID: QmUWmZnpZb6xFryNDeNU7KcJ1Af5oHy7fB9npU67sseEjR, targetAddr: /dns4/stats-infra.kittyhawk.wtf/tcp/8080 heartbeat.go:245

why I fail to send hello to Qm**EjR,but open stream successfully?

**Comment by carsonfly** [comment]
> Can you provide the output of `go-filecoin inspect all`?
Version: 81e75fd

Runtime
OS: linux
Arch: amd64
Version: go1.12.1
Compiler: gc
NumProc: 4
GoMaxProcs: 4
NumGoRoutines: 98
NumCGoCalls: 4

Disk
Free: 8706824503296
Total: 8267494649856
FSType: 61267

Memory
Swap: 0
Virtual: 9520944000

Environment
FilAPI:
FilPath:
GoPath:

Config
{
"api": {
"address": "/ip4/127.0.0.1/tcp/3453",
"accessControlAllowOrigin": [
"http://localhost:8080",
"https://localhost:8080",
"http://127.0.0.1:8080",
"https://127.0.0.1:8080"
],
"accessControlAllowCredentials": false,
"accessControlAllowMethods": [
"GET",
"POST",
"PUT"
]
},
"bootstrap": {
"addresses": [
"/dns4/user.kittyhawk.wtf/tcp/9000/ipfs/Qmd6xrWYHsxivfakYRy6MszTpuAiEoFbgE1LWw4EvwBpp4",
"/dns4/user.kittyhawk.wtf/tcp/9001/ipfs/QmXq6XEYeEmUzBFuuKbVEGgxEpVD4xbSkG2Rhek6zkFMp4",
"/dns4/user.kittyhawk.wtf/tcp/9002/ipfs/QmXhxqTKzBKHA5FcMuiKZv8YaMPwpbKGXHRVZcFB2DX9XY",
"/dns4/user.kittyhawk.wtf/tcp/9003/ipfs/QmZGDLdQLUTi7uYTNavKwCd7SBc5KMfxzWxAyvqRQvwuiV",
"/dns4/user.kittyhawk.wtf/tcp/9004/ipfs/QmZRnwmCjyNHgeNDiyT8mXRtGhP6uSzgHtrozc42crmVbg"
],
"minPeerThreshold": 1,
"period": "10s"
},
"datastore": {
"type": "badgerds",
"path": "badger"
},
"heartbeat": {
"beatTarget": "/dns4/stats-infra.kittyhawk.wtf/tcp/8080/ipfs/QmUWmZnpZb6xFryNDeNU7KcJ1Af5oHy7fB9npU67sseEjR",
"beatPeriod": "3s",
"reconnectPeriod": "10s",
"nickname": "Carson"
},
"mining": {
"minerAddress": "empty",
"autoSealIntervalSeconds": 120,
"storagePrice": "0"
},
"mpool": {
"maxPoolSize": 10000,
"maxNonceGap": "100"
},
"net": "devnet-user",
"observability": {
"metrics": {
"prometheusEnabled": false,
"reportInterval": "5s",
"prometheusEndpoint": "/ip4/0.0.0.0/tcp/9400"
},
"tracing": {
"jaegerTracingEnabled": false,
"probabilitySampler": 1,
"jaegerEndpoint": "http://localhost:14268/api/traces"
}
},
"sectorbase": {
"rootdir": ""
},
"swarm": {
"address": "/ip4/0.0.0.0/tcp/6000"
},
"wallet": {
"defaultAddress": "t1zhfflggti7xaavdunzqe73dijhpukzd256tobvi"
}
}

**Comment by muronglaowang** [comment]
> > 你能提供输出`go-filecoin inspect all`吗？
> > 版本：[81e75fd](https://github.com/filecoin-project/go-filecoin/commit/81e75fd7f9722f9a100b8d03852e4e823a4b209d)
> 
> 运行时
> 操作系统：linux 
> Arch：amd64 
> 版本：go1.12.1 
> 编译器：gc 
> NumProc：4 
> GoMaxProcs：4 
> NumGoRoutines：98 
> NumCGoCalls：4
> 
> 磁盘
> 空闲：8706824503296 
> 总计：8267494649856 
> FSType：61267
> 
> 内存
> 交换：0 
> 虚拟：9520944000
> 
> 环境
> FilAPI：
> FilPath：
> GoPath：
> 
> 配置
> { 
> “api”：{ 
> “address”：“/ ip4 / 127.0.0.1 / tcp / 3453”，
> “accessControlAllowOrigin”：[ 
> “http：// localhost：8080”，
> “https：// localhost：8080”，
> “ http://127.0.0.1:8080" ，
> “https://127.0.0.1:8080” 
> ]，
> “accessControlAllowCredentials”：假，
> “accessControlAllowMethods”：[ 
> “GET”，
> “POST”，
> “PUT” 
> ] 
> }， 
> “ bootstrap“：{ 
> ”addresses“：[ 
> ”/ dns4 / 
> user.kittyhawk.wtf/tcp/9000/ipfs/Qmd6xrWYHsxivfakYRy6MszTpuAiEoFbgE1LWw4EvwBpp4“，”/ dns4 / user.kittyhawk。跆拳道/ TCP / 9001 / IPF问题/ QmXq6XEYeEmUzBFuuKbVEGgxEpVD4xbSkG2Rhek6zkFMp4"
> “/dns4/user.kittyhawk.wtf/tcp/9002/ipfs/QmXhxqTKzBKHA5FcMuiKZv8YaMPwpbKGXHRVZcFB2DX9XY”，
> “/dns4/user.kittyhawk.wtf/tcp/9003/ipfs/QmZGDLdQLUTi7uYTNavKwCd7SBc5KMfxzWxAyvqRQvwuiV”，
> “/dns4/user.kittyhawk.wtf/tcp/ 9004 / ipfs / QmZRnwmCjyNHgeNDiyT8mXRtGhP6uSzgHtrozc42crmVbg“ 
> ]，
> ”minPeerThreshold“：1，
> ”period“：”10s“ 
> }，
> ”datastore“：{ 
> ”type“：”badgerds“，
> ”path“：”badger“ 
> }，
> ”heartbeat“：{ 
> “beatTarget”：“/ dns4 /stats-infra.kittyhawk.wtf / tcp / 8080 / ipfs / QmUWmZnpZb6xFryNDeNU7KcJ1Af5oHy7fB9npU67sseEjR”，
> “beatPeriod”：“3s”，
> “reconnectPeriod“：”10s“， 
> ”nickname“：”Carson“ 
> }，
> ”mining“：{ 
> ”minerAddress“：”empty“，
> “autoSealIntervalSeconds”：120，
> “storagePrice”：“0” 
> }，
> “mpool”：{ 
> “maxPoolSize”：10000，
> “maxNonceGap”：“100” 
> }，
> “net”：“devnet-user”，
> “observability”：{ 
> “metrics”：{ 
> “prometheusEnabled”：false，
> “reportInterval”：“5s”，
> “prometheusEndpoint”：“/ ip4 / 0.0.0.0 / tcp / 9400” 
> }，
> “tracing”：{ 
> “jaegerTracingEnabled”：false，
> “probabilitySampler “：1，
> ”jaegerEndpoint“：” [http：// localhost：14268 / api / traces](http://localhost:14268/api/traces) “ 
> } 
> }，
> “sectorbase”：{ 
> “rootdir”：“” 
> }，
> “swarm”：{ 
> “address”：“/ ip4 / 0.0.0.0 / tcp / 6000” 
> }，
> “wallet”：{ 
> “defaultAddress”：“t1zhfflggti7xaavdunzqe73dijhpukzd256tobvi” 
> } 
> }

You must promise to provide the storage sector after the fil COINS are mortgaged，like this   go-filecoin miner create 10 100 --gas-price = 0.001 --gas-limit = 300 --peerid go-filecoin id | jq -r '.ID'

**Comment by muronglaowang** [comment]
You have to promise the sector that will provide the storage service before the mortgage,like  this :
go-filecoin miner create 10  100 --gas-price=0.001 --gas-limit=300 --peerid go-filecoin id | jq -r '.ID'

**Comment by carsonfly** [comment]
> go-filecoin miner create 10 100 --gas-price=0.001 --gas-limit=300 --peerid go-filecoin id | jq -r '.ID'

root@carsonflytw3:/home/user/filecoin# go-filecoin miner create 10 100 --gas-price=0.001 --gas-limit=300 --peerid go-filecoin id | jq -r '.ID'
Error: Unknown Command "10"

**Comment by carsonfly** [comment]
> You have to promise the sector that will provide the storage service before the mortgage,like this :
> go-filecoin miner create 10 100 --gas-price=0.001 --gas-limit=300 --peerid go-filecoin id | jq -r '.ID'

It still not work

**Comment by travisperson** [comment]
It looks like you are trying to connect to the user devnet. To do so you will need to use the `0.2.4` release which can be found here: https://github.com/filecoin-project/go-filecoin/releases/tag/0.2.4.

The `0.3.0` release is currently on the test devnet getting ready to be released to the user devnet, hopefully sometime next week.

**Comment by muronglaowang** [comment]
Your wallet balance is zero! so you can't take out a mortgage,
Let's look forward to the next version

**Comment by muronglaowang** [comment]
Your wallet has a zero balance,and your block height is not synchronized to the latest block height,so,you can't create miner

**Comment by carsonfly** [comment]
> It looks like you are trying to connect to the user devnet. To do so you will need to use the `0.2.4` release which can be found here: https://github.com/filecoin-project/go-filecoin/releases/tag/0.2.4.
> 
> The `0.3.0` release is currently on the test devnet getting ready to be released to the user devnet, hopefully sometime next week.

I have tried Ver0.24,but still failed,when I type "go-filecoin message wait ${MESSAGE_CID}",no thing return.I confirm the filecoin daemon thread is running .My wallet is always empty.Is the faucet closed?

**Comment by carsonfly** [comment]
> > It looks like you are trying to connect to the user devnet. To do so you will need to use the `0.2.4` release which can be found here: https://github.com/filecoin-project/go-filecoin/releases/tag/0.2.4.
> > The `0.3.0` release is currently on the test devnet getting ready to be released to the user devnet, hopefully sometime next week.
> 
> I have tried Ver0.24,but still failed,when I type "go-filecoin message wait ${MESSAGE_CID}",no thing return.I confirm the filecoin daemon thread is running .My wallet is always empty.Is the faucet closed?

When I type "watch -n 10 'go-filecoin show block $(go-filecoin chain head | head -n 1)'",it return
**
Every 10.0s: go-filecoin show block $(go-filecoin chain head | head -n 1)                                                                                                                                            Mon Jul  8 09:17:33 2019

Block Details
Miner:  empty
Weight: 0.000
Height: 0
Nonce:  0
**
It seems something wrong.how can I solve it.

**Comment by travisperson** [comment]
Chain syncing is currently a two step process. The node first downloads the chain, and then verifies it by running it through consensus. The `chain head` result will not change until consensus starts to run which may take a while depending on how well connected you are to other node on the network, and how quickly you can download the chain. The current chain on 0.2.4 is ~27,000 blocks.

You should see logs such as the following which will tell you how far along the process you are of downloading the blocks.

```
19:13:21.902  INFO chain.sync: fetching the chain, 500 blocks fetched default_syncer.go:151
19:13:31.036  INFO chain.sync: fetching the chain, 1000 blocks fetched default_syncer.go:151
19:13:39.632  INFO chain.sync: fetching the chain, 1500 blocks fetched default_syncer.go:151                                                                                                                                                                                                                                                                                                    
19:13:49.843  INFO chain.sync: fetching the chain, 2000 blocks fetched default_syncer.go:151                                                                                                                                                                                                                                                                          
19:13:59.606  INFO chain.sync: fetching the chain, 2500 blocks fetched default_syncer.go:151 
```

**Comment by carsonfly** [comment]
> Chain syncing is currently a two step process. The node first downloads the chain, and then verifies it by running it through consensus. The `chain head` result will not change until consensus starts to run which may take a while depending on how well connected you are to other node on the network, and how quickly you can download the chain. The current chain on 0.2.4 is ~27,000 blocks.
> 
> You should see logs such as the following which will tell you how far along the process you are of downloading the blocks.
> 
> ```
> 19:13:21.902  INFO chain.sync: fetching the chain, 500 blocks fetched default_syncer.go:151
> 19:13:31.036  INFO chain.sync: fetching the chain, 1000 blocks fetched default_syncer.go:151
> 19:13:39.632  INFO chain.sync: fetching the chain, 1500 blocks fetched default_syncer.go:151                                                                                                                                                                                                                                                                                                    
> 19:13:49.843  INFO chain.sync: fetching the chain, 2000 blocks fetched default_syncer.go:151                                                                                                                                                                                                                                                                          
> 19:13:59.606  INFO chain.sync: fetching the chain, 2500 blocks fetched default_syncer.go:151 
> ```
How can I view this log?What the command can I use to view the log?please help me

**Comment by carsonfly** [comment]
I meet  a problem,when I  am syncing and validating the existing blockchain, but it stop at block height 9939.I restart the daemon thread some times,but the problem still exist.
Every 10.0s: go-filecoin show block $(go-filecoin chain head | head -n 1)                                                Wed Jul 10 03:45:27 2019

Block Details
Miner:  t25k3dvjwsj5647ii3kkexfqvfff6j44k3ty24i7y
Weight: 206023.874
Height: 9939
Nonce:  0

**Comment by anorth** [comment]
@acruikshank the original issue reported here is relevant to our recent discussions, though it has now devolved into chain syncing problems.

@carsonfly we recently released a new 0.3.2 build and have reset the user devnet. Please fetch the new code and try again. The new chain only has ~2400 blocks at present.

**Comment by anorth** [comment]
A new 0.4 release will be out soon, with new chain sync. I'm going to close this issue, but please re-open, or open a new one, if you experience new problems.
